### PR TITLE
[vectorized](bug) fix some case in enable fold constant

### PR DESCRIPTION
**Pull request by zhangstar333**
# Proposed changes

some case will be failed in open fold constant:
`SELECT murmur_hash3_64("hello")`
`select cast(NULL as struct<f1:string,f2:int>)`
`select repeat("a",3)`
`select cast('{}' as struct<f1:int>);`

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/literal/Literal.java [hunk 1]
```diff
@@ -236,7 +236,7 @@ protected Expression uncheckedCastTo(DataType targetType) throws AnalysisExcepti
         } else if (targetType.isIntegerType()) {
             return Literal.of(Double.valueOf(desc).intValue());
         } else if (targetType.isBigIntType()) {
-            return Literal.of(Double.valueOf(desc).longValue());
+            return Literal.of(new BigInteger(desc).longValue());
```
Comment (HappenLee) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/literal/Literal.java:239]: Long.valueof() ?

**Comment by zhangstar333** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 33.7 seconds
 stream load tsv:          454 seconds loaded 74807831229 Bytes, about 157 MB/s
 stream load json:         24 seconds loaded 2358488459 Bytes, about 93 MB/s
 stream load orc:          73 seconds loaded 1101869774 Bytes, about 14 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230323063134_clickbench_pr_119317.html

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

**Review by HappenLee — DISMISSED** [review]
LGTM

**Review by BiteTheDDDDt — APPROVED** [review]
LGTM

**Pull request merged**